pkg/admin/services/prometheus_service_impl.go

```diff
@@ -39,6 +40,37 @@ var (
 
 type PrometheusServiceImpl struct{}
 
+func (p *PrometheusServiceImpl) PromDiscovery(w http.ResponseWriter) error {
+	w.Header().Set("Content-Type", "application/json")
+	prosvc, err := providerService.FindService(constant.IP, constant.AnyValue)
```

Comment:
Would directly calling on `providerService.findAddresses()` be better?

Reply:
Similar issue for the consumer-side address computation process. It would be best to call a method that operates on `InterfaceRegistryCache` and returns URL list directly to seek performance.